fixes #13932, #11627: Updated instruction for a working Cross-compile to Windows from Linux using the latest version of MXE

remyroy opened the pull request:
Closes #13932, closes #11627.

The current instructions to cross-compile to Windows from Linux are not working. This PR updates these instruction to obtain a fully working build.

~~The latest version of MXE seems to have some problem with [their sdl2_image and freetype packages](https://github.com/mxe/mxe/issues/970). I'll update this PR once that is resolved.~~

This updated guide using the latest version of MXE works and has been tested on Ubuntu 15.04. It produce proper and working builds for both 32-bit and 64-bit Windows which have been tested on Windows 7 64-bit.

remyroy commented:
I have had to change the LDFLAGS in Makefile for the latest MXE version to work properly. This might break the Jerkins builds. What do you think @narc0tiq ?

It might be the time to update MXE to the latest version on Jerkins since I believe is it using [a version from Mar 19, 2014](https://github.com/mxe/mxe/commits/9582b46).

narc0tiq commented:
My test build built just fine, even with the old mxe, though this does also mean I can upgrade safely now. Thanks!

BevapDin commented:
It might be worth mentioning the value for `CROSS=` should be the path to whatever g++ one wants to use *minus* the `g++` part at the end. 

I imagine that many people don't know the exact "PLATFORM" string, but will be able to locate the g++ binary, thereby providing an easy way to determine the value for "CROSS".

remyroy commented:
I moved the linux native sections together in the COMPILING.md document. It makes more sense to have them be together than to have the SDL from different platform together.

I've added details on the CROSS flag as well.

kevingranade commented:
FYI this broke the build bot so I reverted the Makefile change.

remyroy commented:
Where can I find the failed build log? I might be able to have a the old version of MXE on the build bot and the latest one working at the same time.

remyroy commented:
I see 2 failed build on jerkins that failed because the Makefile changed. ([First](http://ci.narc.ro/job/Cataclysm-Jently/36759/console), [Second](http://ci.narc.ro/job/Cataclysm-Jently/36760/console))

@narc0tiq told me that he has protection against potential attacks against malicious code inserted in the Makefile by comparing it with a reference file. In order to have the build bot accept changes to the Makefile, it might be necessary to manually update that reference file.

There is also a [third failed build](http://ci.narc.ro/job/Cataclysm-Jently/36754/console) related to a compilation error in `src/game.cpp` which I guess is unrelated to this change.

remyroy commented:
I found the culpit log in http://ci.narc.ro/job/Cataclysm-Matrix/Graphics=Tiles,Platform=Windows/3897/console. I'll see if I can work on a fix.

narc0tiq commented:
> protection against potential attacks against malicious code inserted in the Makefile by comparing it with a reference file. 

Which applies only to the pull request builds. The commits that make it into master are considered trustworthy, as they will have been vetted by at least one person with push rights. Indeed, the builds _of_ master are the ones that create the reference Makefile in the first place.

remyroy commented:
#14040 should complete this PR. It should hopefully work on the build bot as well.
